packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -99,8 +182,41 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
 
   void _flushPointerEventQueue() {
     assert(!locked);
-    while (_pendingPointerEvents.isNotEmpty)
-      _handlePointerEvent(_pendingPointerEvents.removeFirst());
+
+    // Stop resampler if resampling is not enabled. This is a no-op if
+    // resampling was never enabled.
+    if (!resamplingEnabled) {
+      _resampler.stop(_handlePointerEvent);
+    }
+
```

Comment:
Can everything below also get encapsulated inside `_Resampler`?

For example, instead of calling `_resampler.addEvent(event)`, we can send in a list of events `_resampler.addEvents(_pendingPointerEvents)`. The `_resampler` can also save the callback `_handlePointerEvent` as its field so it can schedule a sample callback by its own?

Reply:
We would fail to detect when resampling is disabled in that case but we could introduce some callback cancel logic to handle that. We would also need some wrapper to ensure that we're not violating the "if (!locked)" condition. The overall complexity might increase but if encapsulation is highest priority then it might still make sense. Wdyt?

Reply:
I did this additional encapsulation in latest version of this CL